### DISCUSSION TITLE
Routing all audio sources (USB in, Spotify, ...) to different output (channels 1/2, channel 3/4) of external audio interface

### DISCUSSION THREAD

**Discussion by gregvds** [ExistentialAudio/BlackHole#706] [Q&A]
Hi everyone,

I would like to have all my audio sources (spotify, USB compliant external synth audio in, etc) to be routed to the audio out channels of an external USB audio interface (RME ADI-2 Pro FS R BE).

I would like to treat USB interface output channel 3/4 through Sonarworks SoundID Reference standalone app (no DAW here), and leave channel 1/2 untreated, so all the sounds produced by MacOS, or received from external USB instruments would be corrected for Headphones 3/4 output of the RME audio interface, while the RME channels 1/2 stay untreated, to be heard on active monitors.

Is it possible with a combination of Blackhole, aggregate devices and multiple output device, or there is no alternative to Rogue Amoeba Loopback for this kind or routing?

Thank you already,

Greg

**Comment by devinroth** [comment]
What have you tried?

Devin

**Comment by gregvds** [comment]
Sadly, I still don't have the RME ADI-2, so I cannot truely try this.

I just played a bit around with blackhole 2ch, trying to use it as the treated output in SoundID, but that did not produced any sound, for SoundID is its own virtual audio bridge device that put itself between the given output and the true output of the Mac (explanations received from Sonarworks support).

So, currently, my question was more a possible use case, more than a 'I'm stuck, how can I actually have it working'.

I should try and draw some block diagrams/flow chart, to have a clearer idea of how things should be organized, and if blackhole can fit inside it.

Sorry for the vague question, I know it's not easy to help out someone that comes with this type of question...

Thank you anyway for the development and the solution for all of us outhere :-)

**Comment by devinroth** [comment]
BlackHole is a loopback driver. Audio Output -> Audio Input.

Other tools you can use is the aggregate device which combines audio devices into a single one, multi-output which outputs audio to all devices on the list, you can use DAWS to route audio from any input to any output. My bet is you could make something work with an aggregate and a DAW. Hope that help.

Devin
